src/lib.rs

```diff
@@ -55,6 +57,15 @@ pub enum Error {
     StatusError(Status),
     /// Device state is in error: {0}
     StateError(State),
+    /// Incompatible DFU protocol
+    InCompatibleProtocol,
```

Comment:
The casing is wrong. This is actually two words. not three. 
```suggestion
    IncompatibleProtocol,
```